erigon-lib/state/files_item.go

```diff
@@ -63,9 +63,13 @@ type filesItem struct {
 }
 
 func newFilesItem(startTxNum, endTxNum, stepSize uint64) *filesItem {
+	return newFilesItemWithFrozenSteps(startTxNum, endTxNum, stepSize, config3.StepsInFrozenFile)
+}
+
+func newFilesItemWithFrozenSteps(startTxNum, endTxNum, stepSize uint64, stepsInFrozenFile uint64) *filesItem {
 	startStep := startTxNum / stepSize
 	endStep := endTxNum / stepSize
-	frozen := endStep-startStep == config3.StepsInFrozenFile
+	frozen := endStep-startStep >= stepsInFrozenFile
```

Comment:
can you double-check: `caplin's` files are sharded by various params and with various freezing limit. do we mark old Caplin's files as `frozen`?

Reply:
caplin is not merging anything...it has snapshots of two ranges: ERigon2MergeLimit (earlier value for snapshot size); and now it's changed to CaplinMergeLimit...both are considered "frozen" (and rightly so).

Reply:
Tnx